library/core/src/ptr/non_null.rs

```diff
@@ -1803,4 +1811,67 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    #[kani::proof_for_contract(NonNull::sub)]
+    pub fn non_null_check_sub() {
+        const SIZE: usize = 10;
+        // Randomiz pointer offset within array bound
```

Comment:
Can you use the `PointerGenerator` functionality for your harnesses instead? I think it will be cleaner -- see [here](https://github.com/model-checking/verify-rust-std/discussions/162).